### ISSUE TITLE
nrunner: avocado is lefting processes behind after a kill

### ISSUE THREAD

**Issue by beraldoleal** [avocado-framework/avocado#4911]
After #7ffd0fdd92fc6e3f7a42a9983d0dd259ec0674ec avocado is failing this test:

```bash
$ export AVOCADO_CHECK_LEVEL=3
$ avocado run ./selftests/functional/test_interrupt.py:InterruptTest.test_well_behaved_sigint

JOB ID     : c7a7b763666f03a9f5f194111194be2396d1e787
JOB LOG    : /home/local/avocado/job-results/job-2021-08-28T17.09-c7a7b76/job.log
 (1/1) ./selftests/functional/test_interrupt.py:InterruptTest.test_well_behaved_sigint: STARTED
 (1/1) ./selftests/functional/test_interrupt.py:InterruptTest.test_well_behaved_sigint: FAIL (12.45 s)
RESULTS    : PASS 0 | ERROR 0 | FAIL 1 | SKIP 0 | WARN 0 | INTERRUPT 0 | CANCEL 0
JOB HTML   : /home/local/avocado/job-results/job-2021-08-28T17.09-c7a7b76/results.html
JOB TIME   : 13.24 s

```

**Comment by clebergnu** [comment]
@beraldoleal for Milestone 91, we better switch that test to use `runner` instead of `nrunner`.

**Comment by clebergnu** [comment]
IMO this issue should be closed, given that the solution with the nrunner requires not a fix, but a lot of infrastructure (according to description in #4994, added to sprint 93).

**Comment by ana** [comment]
Closing in favour of https://github.com/avocado-framework/avocado/issues/4994
